OpenRA.Game/Map/CellLayer.cs

```diff
@@ -98,6 +98,25 @@ int Index(MPos uv)
 			}
 		}
 
+		public bool TryGetValue(CPos cell, out T value)
+		{
+			if (GridType == MapGridType.RectangularIsometric && cell.X < cell.Y)
```

Comment:
Can we please also include the comment from `Contains` explaining the `X < Y` check?